middleware.go

```diff
@@ -4,22 +4,19 @@ import (
 	"net/http"
 )
 
-// NextMiddlewareFunc is the function signature for the next parameter in ServerHTTPMiddleware
-type NextMiddlewareFunc func(http.ResponseWriter, *http.Request)
```

Comment:
Try [this](https://play.golang.org/p/xi245_WCCg)
Unless I'm doing something silly I can't get a function that doesn't use the convenience wrapper to play nice with an interface that expects it